railgun/pkg/config/controller_configuration.go

```diff
@@ -0,0 +1,79 @@
+package config
+
+import (
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	knativev1alpha1 "knative.dev/networking/pkg/apis/networking/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/kong/kubernetes-ingress-controller/pkg/util"
+	konghqcomv1 "github.com/kong/kubernetes-ingress-controller/railgun/apis/configuration/v1"
+	"github.com/kong/kubernetes-ingress-controller/railgun/internal/ctrlutils"
+)
+
+// -----------------------------------------------------------------------------
+// Controller Configurations
+// -----------------------------------------------------------------------------
+
+// ControllerConfig collects the enablement status of controllers which indicates
+// to the controller manager which and which not to run during initialization.
+type ControllerConfig struct {
```

Comment:
Generically, when we define struct, we also define function against this struct. i.e

func (ctrlconfig *ControllerConfig) GetKnativeEnableStatus() bool{
}

otherwise, maybe you just want to re-use existing const

Reply:
Having a struct type and methods on it is not mutally exclusive.